options_test.go

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package ringpop
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+	"github.com/uber/ringpop-go/testutils/mocks"
+	"github.com/uber/tchannel-go"
+)
+
+type RingpopOptionsTestSuite struct {
+	suite.Suite
+	ringpop *Ringpop
+	channel *tchannel.Channel
+}
+
+func (s *RingpopOptionsTestSuite) SetupTest() {
+
+	ch, err := tchannel.NewChannel("test", nil)
+	s.Require().NoError(err, "Channel creation failed")
+
+	s.channel = ch
+}
+
+// TestDefaults tests that the default options are applied to a Ringpop
+// instance when none are specified by the user.
+func (s *RingpopOptionsTestSuite) TestDefaults() {
+
+	rp, err := New("test", Channel(s.channel))
+	s.Require().NotNil(rp)
+	s.Require().NoError(err)
+
+	s.NotNil(rp.logger)
```

Comment:
shall we also assert that `rp.logger` and `rp.statter` are equal to the default values?